src/tools/blast/components/BlastForm.tsx

```diff
@@ -278,6 +283,7 @@ const BlastForm = () => {
       gapped: gapped.selected as GapAlign,
       // transform string into number
       hits: parseInt(hits.selected as string, 10) as Scores,
+      hsps: hsps.selected as HSPs,
```

Comment:
```suggestion
      hsps: parseInt(hsps.selected, 10) || undefined as HSPs,
```
This string "undefined" is passed to the job parameters resulting  in a 400:
```
<?xml version='1.0' encoding='UTF-8'?>
<error>
 <description>For input string: "undefined"</description>
</error>
```

Reply:
Mhhh, that's strange, I'll have a look

Reply:
my previous suggestion had type issues
```suggestion
      hsps: (hsps.selected === 'undefined'
        ? undefined
        : parseInt(hsps.selected as string, 10)) as HSPs,
```

Reply:
third times a charm 😅 
```suggestion
      hsps: (parseInt(hsps.selected as string, 10) || undefined) as HSPs,
```

Reply:
All good now, good catch, I hadn't try going back and forth between options when testing, and thanks for the suggestion